### PR TITLE
Change var twice in on_change (#286)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
State variable will be updated but the on_change will only be called the first time

### CODE REVIEW

File: taipy/gui/gui.py [hunk 1]
```diff
@@ -272,10 +269,41 @@ def __set_client_id_in_context(self, client_id: t.Optional[str] = None):
...
+            der_vars.update(derived_vars),
+        if var_name in modified_vars:
+            return True
+        else:
+            modified_vars.add(var_name)
+            return False
+
+    def __clean_vars_on_exit(self) -> t.Optional[t.Set[str]]:
+        update_count = getattr(g, "update_count", 0) - 1
+        print(f"__clean_vars_on_exit: {update_count} \n\tmodified_vars: {getattr(g, 'modified_vars', None)} \n\tderived_vars: {getattr(g, 'derived_vars', None)}")
```
Comment (FabienLelaquais) [inline taipy/gui/gui.py]: Are you sure you want to keep this??

**Review by FabienLelaquais — APPROVED** [review]
I... think I got it.

File: taipy/gui/gui.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui/gui.py]: oups 👎

**Review by dinhlongviolin1 — APPROVED** [review]
I knew this would need to be dealt with as soon as we use g. Well done Fred!

**Pull request merged**